src/ValueObject/EmailAddress.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace ThePay\ApiClient\ValueObject;
+
+use Egulias\EmailValidator\EmailValidator;
+use Egulias\EmailValidator\Validation\RFCValidation;
+
+class EmailAddress extends NonEmptyString
+{
+    public static function filter($value)
+    {
+        $nonEmptyString = parent::filter($value);
+
+        if ((new EmailValidator())->isValid($nonEmptyString, new RFCValidation()) === false) {
```

Comment:
We preparing break change version, so we can also add validation for valid DNS in email: `Egulias\EmailValidator\Validation\DNSCheckValidation`.

Example:
`$emailValidator->isValid($value, new MultipleValidationWithAnd([new RFCValidation(), new DNSCheckValidation()])`

Reply:
It required **changes in some tests and documentation** and I also modified the error message where I changed _"e-mail address"_ to _"public e-mail address"_ because **non-public and well known test domains do not pass** validation now.